app/src/main/contrib/components/Responsibilities/Overview/ConfidentialityAgreement.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from "react";
+import {settings} from "../../../../shared/Settings";
+import {connect} from "react-redux";
+import {branch, compose, renderNothing} from "recompose";
+import {Dispatch} from "redux";
+import {ContribAppState} from "../../../reducers/contribAppReducers";
+import {default as withLifecycle, LifecycleMethods} from "@hocs/with-lifecycle";
+import {userActionCreators} from "../../../actions/userActionCreators";
+
+export interface ConfidentialityAgreementPublicProps {
+    touchstoneId: string;
+}
+
+export interface ConfidentialityAgreementProps extends ConfidentialityAgreementPublicProps {
+    signAgreement: () => void;
+    signed: boolean;
+    getConfidentiality: () => void;
+}
+
+const fullConfidentialityAgreement = require('./rfp-applicants-confidentiality.pdf');
+
+export const ConfidentialityAgreementComponent: React.SFC<ConfidentialityAgreementProps>
+    = (props: ConfidentialityAgreementProps) => {
+
+    if (settings.isApplicantTouchstone(props.touchstoneId)) {
+        return <div className={"row"}>
+            <div className={"col-12 col-md-6 offset-md-3"}>
+                <div className={"border p-3 border-dark mb-5"}>
+                    I have read and understood the terms of the <a href={fullConfidentialityAgreement}>
+                    RfP applicants' confidentiality agreement</a>. In doing so, I understand and agree not to disclose
+                    or share any information on vaccine coverage data which I access from Montagu,
+                    beyond my immediate RfP modelling group.
+                    {!props.signed &&
+                    <input type={"checkbox"} className={"mt-2 mb-2 d-block"}
+                           onChange={props.signAgreement} style={{height: "20px", width: "20px"}}/>
+                    }
+                </div>
+            </div>
+        </div>
+    }
+};
+
+const lifecyleProps: Partial<LifecycleMethods<ConfidentialityAgreementProps>> = {
+    onDidMount(props: ConfidentialityAgreementProps) {
+        props.getConfidentiality();
```

Comment:
Presumably there is a delay here while this is fetched. Until then, regardless of whether or not the user has signed before won't they be presented with this checkbox?

Reply:
Yes, in practice this is a very short delay, but you're right. Have changed default state of the `signed` variable to null, so this only gets rendered once the value has been fetched.